### PR TITLE
Docker build container and usename change

### PR DESCRIPTION
**Pull request by ppankd**
Added a build container
Command to create a comdb2 image is 
make docker
This creates a build container and creates a comdb2 container
Changes to read usenames from comdb2 global file.

### CODE REVIEW

**Comment by mponomar** [comment]
Pankaj, see also #37

**Comment by ppankd** [comment]
My apologies. I have fixed the makefile changes which had made it docker build specific. Please let me know if this same PR can be used.

**Comment by riverszhang89** [comment]
@ppankd That's all right. We all make mistakes and that's why we have created @roborivers to run regression testing for every pull request.

**Comment by akshatsikarwar** [comment]
Robo Rivers is a force of nature

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM artprod.dev.bloomberg.com/bfs/base-ubuntu:xenial
```
Comment (mponomar) [inline contrib/docker/Dockerfile]: This should probably just be ubuntu:xenial?

File: db/comdb2.c [hunk 2]
```diff
@@ -2241,7 +2241,9 @@ static int read_lrl_option(struct dbenv *dbenv, char *line, void *p, int len)
         /* Copycomdb2 still wants this option, but it's useless for anything else.
            Ignore. */
     } else if (tokcmp(tok, ltok, "usenames") == 0) {
-        /* Also ignore - we treat this in pre_read_option */
+        /* This was preventing setting usenames at a global level
```
Comment (mponomar) [inline db/comdb2.c]: This should definitely go in.

**Comment by mponomar** [comment]
Thanks.  We have a couple of pull requests for Docker support now.  We're going to combine them, and get that in.

**Comment by mponomar** [comment]
Merged this to #37, awaiting review.

**Pull request closed without merging**